### PR TITLE
feat: Liquid support

### PR DESCRIPTION
**Pull request by dni**
first steps for liquid support

- update to embit 0.7.0
- add pair to BoltzClient constructor
- add `get-pairs` cli command

### CODE REVIEW

File: boltz_client/cli.py [hunk 1]
```diff
@@ -1,24 +1,31 @@
...
-# )
+config = BoltzConfig(
+    pairs=["BTC/BTC", "L-BTC/BTC"],
+    network="regtest",
+    network_liquid="elementsregtest",
+    api_url="http://localhost:9001",
+    mempool_url="http://localhost:8999/api/v1",
+    mempool_ws_url="ws://localhost:8999/api/v1/ws",
+    mempool_liquid_url="http://localhost:8998/api/v1",
+    mempool_liquid_ws_url="ws://localhost:8998/api/v1/ws",
```
Comment (jackstar12) [inline boltz_client/cli.py:27]: `ws_url` could be derived from the normal `url`

File: boltz_client/mempool.py [hunk 2]
```diff
@@ -15,7 +15,9 @@
 @dataclass
 class LockupData:
     status: str
+    tx: str
```
Comment (jackstar12) [inline boltz_client/mempool.py:18]:
```suggestion
    tx_hex: str
```

**Review by jackstar12 — APPROVED** [review]

File: boltz_client/boltz.py [hunk 3]
```diff
@@ -1,12 +1,13 @@
...
 
 import asyncio
-from dataclasses import dataclass
+from dataclasses import dataclass, field
 from enum import Enum
 from math import ceil, floor
 from typing import Optional
 
 import httpx
+from embit.liquid.addresses import to_unconfidential
```
Comment (michael1011) [inline boltz_client/boltz.py:10]: Why are you still using embit for liquid here?

File: boltz_client/boltz.py [hunk 4]
```diff
@@ -261,33 +311,49 @@ async def refund_swap(
...
+
+        if self.pair == "L-BTC/BTC":
+            unconfidential_lockup_address = to_unconfidential(lockup_address)
+            if not unconfidential_lockup_address:
+                raise BoltzApiException("can not unconfidentialize lockup address")
+            lockup_address = unconfidential_lockup_address
+            unconfidential_receive_address = to_unconfidential(receive_address)
+            if not unconfidential_receive_address:
+                raise BoltzApiException("can not unconfidentialize receive address")
+            self.validate_address(unconfidential_receive_address)
```
Comment (michael1011) [inline boltz_client/boltz.py:326]: Code duplication from `claim_reverse_swap`

File: boltz_client/onchain.py [hunk 5]
```diff
@@ -49,12 +64,14 @@ def create_refund_tx(
     redeem_script_hex: str,
     timeout_block_height: int,
     lockup_tx: LockupData,
+    pair: str,
     fees: int,
-) -> tuple[str, str]:
+    blinding_key: Optional[str] = None,
+) -> str:
     # encrypt redeemscript to script_sig
```
Comment (michael1011) [inline boltz_client/onchain.py:71]: It's not encryption

File: boltz_client/onchain_wally.py [hunk 6]
```diff
@@ -0,0 +1,203 @@
...
+from __future__ import annotations
+
+import secrets
+from typing import Optional
+
+from .mempool import LockupData
+
+LASSET = bytes.fromhex(
+    "5ac9f65c0efcc4775e0baec4ec03abdde22473cd3cf33c0419ca290e0751b225"
+)[::-1]
```
Comment (michael1011) [inline boltz_client/onchain_wally.py:16]: Is different on every network

File: boltz_client/onchain_wally.py [hunk 7]
```diff
@@ -0,0 +1,203 @@
...
+    blinding_key_bytes = bytes.fromhex(blinding_key)
+
+    receive_blinding_pubkey = wally.confidential_addr_segwit_to_ec_public_key(
+        receive_address, "el"
+    )  # type: ignore
+    receive_unconfidential_address = wally.confidential_addr_to_addr_segwit(
+        receive_address, "el", "ert"
+    )  # type: ignore
+    receive_script_pubkey = wally.addr_segwit_to_bytes(
+        receive_unconfidential_address, "ert", 0
```
Comment (michael1011) [inline boltz_client/onchain_wally.py:67]: Address prefixes are different for each network too

File: boltz_client/onchain_wally.py [hunk 8]
```diff
@@ -0,0 +1,203 @@
...
+    )  # type: ignore
+
+    # parse lockup tx
+    lockup_transaction = wally.tx_from_hex(
+        lockup_tx.tx, wally.WALLY_TX_FLAG_USE_ELEMENTS
+    )
+    vout_n: Optional[int] = None
+    for vout in range(wally.tx_get_num_outputs(lockup_transaction)):
+        script_out = wally.tx_get_output_script(lockup_transaction, vout)  # type: ignore
+        pub_key = wally.addr_segwit_from_bytes(script_out, "ert", 0)
```
Comment (michael1011) [inline boltz_client/onchain_wally.py:77]: Hardcoded address prefix

**Comment by jackstar12** [comment]
`show-pairs` should pretty print the json

**Pull request merged**